File: TransCoder_tokenized_test_set_functions/java/LONGEST_SUBARRAY_SUM_DIVISIBLE_K.java
static int longSubarrWthSumDivByK ( int arr [ ], int n, int k ) {
  HashMap < Integer, Integer > um = new HashMap < Integer, Integer > ( );
  int mod_arr [ ] = new int [ n ];
  int max = 0;
  int curr_sum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    curr_sum += arr [ i ];
    mod_arr [ i ] = ( ( curr_sum % k ) + k ) % k;
  }
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( mod_arr [ i ] == 0 ) max = i + 1;
    else if ( um . containsKey ( mod_arr [ i ] ) == false ) um . put ( mod_arr [ i ], i );
    else if ( max < ( i - um . get ( mod_arr [ i ] ) ) ) max = i - um . get ( mod_arr [ i ] );
  }
  return max;
}

File: TransCoder_tokenized_test_set_functions/python/LONGEST_SUBARRAY_SUM_DIVISIBLE_K.py
def longSubarrWthSumDivByK ( arr, n, k ) :
    um = {
    i : 0 for i in range ( 8 ) }
    
    mod_arr = [ 0 for i in range ( n ) ]
    max = 0
    curr_sum = 0
    for i in range ( n ) :
        curr_sum += arr [ i ]
        mod_arr [ i ] = ( ( curr_sum % k ) + k ) % k
    for i in range ( n ) :
        if ( mod_arr [ i ] == 0 ) :
            max = i + 1
        elif ( mod_arr [ i ] in um ) :
            um [ mod_arr [ i ] ] = i
        else :
            if ( max < ( i - um [ mod_arr [ i ] ] ) ) :
                max = i - um [ mod_arr [ i ] ]
    return max
